streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -288,6 +291,14 @@ void closeRevokedSuspendedTasks() {
     void shutdown(final boolean clean) {
         final AtomicReference<RuntimeException> firstException = new AtomicReference<>(null);
 
+        if (threadProducer != null) {
```

Comment:
Why we nee to commit again upon shutting down? If we are shutting down cleanly we should have already committed right?